ipv8/messaging/anonymization/hidden_services.py

```diff
@@ -490,7 +490,8 @@ def on_establish_intro(self, source_address, payload, circuit_id):
         if not self.ipv8:
             self.logger.error('No IPv8 service object available, cannot start PEXCommunity')
         elif payload.info_hash not in self.pex:
-            community = PexCommunity(self.my_peer, self.ipv8.endpoint, Network(), info_hash=payload.info_hash)
+            community = PexCommunity(self.my_peer, PexEndpointAdapter(self.ipv8.endpoint), Network(),
```

Comment:
Woops, this adapter should've been one instance level. Moving this back to WIP for a bit.